### ISSUE TITLE
Dimension tools: Add icons for block spacing and min. height

### ISSUE THREAD

**Issue by jasmussen** [WordPress/gutenberg#62618]
The current dimensions panel has icons for padding and margin stepped controls, but not for block spacing and min-height controls. This causes misalignment of the controls, making it less scannable. Additionally, in case of width-based controls, or axial gap configuration, an icon denoting the orientation can help anchor the feature.

Here's a design proposal that adds icons for block spacing and min-height, shown before and after:

<img width="865" alt="before and after" src="https://github.com/WordPress/gutenberg/assets/1204802/026b1e9b-67cc-48a6-87f8-648c08144583">

Beyond lining up vertically, this also unifies the controls so that in the future, axial controls (`row-gap` or `column-gap` vs. just `gap`) could be added to the block spacing control. To that end, here are potential icon designs that would work for axial gap configurations:

<img width="720" alt="dimensions-icons" src="https://github.com/WordPress/gutenberg/assets/1204802/b90f6180-3354-4afd-a1fe-a025c0f4b6c1">

[Figma](https://www.figma.com/design/1lnsBl5YhVC3IlbGii6vCT/Dimensions-Panel?node-id=20-182&t=GTjXaEzpqL1cXKxW-1).

**Comment by jarekmorawski** [comment]
Good idea! I think this is a meaningful improvement, but only when paired with more granular settings for controlling the vertical and horizontal gap separately. I can already think of a use case: in WooCommerce, you may want to add a small gap to the product collection so the rows have more breathing room while the grid's density remains the same. Currently, I'd do that by adding a top padding to the product card block, but this setting would work much better.

We'd make Block Spacing work similarly to Padding and Margin, meaning that we'd surface granular settings when the user clicks the icon next to the label. Here's a quick sketch.

<img width="1050" alt="image" src="https://github.com/WordPress/gutenberg/assets/79307566/e17aacf6-736b-4868-bd11-255f2f480d24">

In the future, this consistency would enable us to easily overhaul the whole Dimensions panel and clean up all UI controls at once: from Margin to Block Spacing to Width.

**Comment by richtabor** [comment]
I'm not sure if it's the right icon, but I do like that it makes the sliders more uniform and understandable.

**Comment by akasunil** [comment]
For min height icon, could we use something like this? 
 <img src="https://github.com/WordPress/gutenberg/assets/17861443/9dbc8c2e-3423-4a8e-8296-f2975902e583" width="48px" height="48px">
 
 originally proposed in https://github.com/WordPress/gutenberg/issues/55734 for full height toggle.

**Comment by mateuswetah** [comment]
Not a big fan of the proposed height icon too... it reminds me to much an "I". I like the bar with arrows... or maybe we could have something close to this, using a bit of the style of the side padding and margin?

![image](https://github.com/WordPress/gutenberg/assets/1184874/4e81bb3b-7424-4db4-85e7-62846e5ff633 )

**Comment by jameskoster** [comment]
A small detail, but it would be good to use the new [DropdownMenu](https://wordpress.github.io/gutenberg/?path=/docs/components-experimental-dropdownmenu-v2--docs) component here:

<img width="242" alt="Screenshot 2024-07-01 at 11 02 58" src="https://github.com/WordPress/gutenberg/assets/846565/ae934e54-37eb-461d-b9a2-9cac02169fb5">

Since this is not a multi-select option we should avoid the checkbox icon and use `DropdownMenuRadioItem`. An additional consideration is that radio menu items do not support an icon. So if the icon is critical, we may need to consider a different UI (segmented control?).
